crates/iota-keys/src/keystore.rs

```diff
@@ -425,62 +523,165 @@ impl FileBasedKeystore {
         })
     }
 
-    pub fn set_path(&mut self, path: &Path) {
-        self.path = path.to_path_buf();
+    fn needs_migration(path: &PathBuf) -> Result<bool, anyhow::Error> {
```

Comment:
This file got quite large now, mind to split it into multiple files?
Maybe something like this
```
keystore/
	mod.rs
	file-based.rs
	in-memory.rs
	migration.rs
```

Reply:
I can do it for sure, but it will make the diff not really readable.
Mind if I do this in a separate PR after these changes are done?

Reply:
Sure